### PR TITLE
build(deps): Update google-github-actions, actions/checkout, and setup-node for appengine/metadata

### PR DESCRIPTION
**Pull request by sofisl**

### CODE REVIEW

**Comment by ace-n** [comment]
N.B: there's a typo in the workflow file's name (`apengine` --> `appengine`)

**Comment by sofisl** [comment]
> N.B: there's a typo in the workflow file's name (`apengine` --> `appengine`)

That's why the tests aren't running! Hehe, thank you!

**Review by ace-n — APPROVED** [review]

File: .github/workflows/appengine-metadata-flexible.yaml [hunk 1]
```diff
@@ -21,30 +21,30 @@ jobs:
       pull-requests: 'write'
       id-token: 'write'
     steps:
-    - uses: 'google-github-actions/auth@v0.3.1'
+    - uses: 'google-github-actions/auth@v0.6.0'
```
Comment (tmatsuo) [inline .github/workflows/appengine-metadata-flexible.yaml:24]: I think you need to checkout before using `auth` action.

**Review by tmatsuo — CHANGES_REQUESTED** [review]

File: .github/workflows/appengine-metadata-flexible.yaml [hunk 2]
```diff
@@ -21,30 +21,30 @@ jobs:
...
       with:
         workload_identity_provider: 'projects/1046198160504/locations/global/workloadIdentityPools/github-actions-pool/providers/github-actions-provider'
         service_account: 'kokoro-system-test@long-door-651.iam.gserviceaccount.com'
         create_credentials_file: 'true'
         credentials_json: '${{ secrets.GOOGLE_CREDENTIALS }}'
         access_token_lifetime: 600s
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
       with:
         ref: ${{github.event.pull_request.head.ref}}
```
Comment (tmatsuo) [inline .github/workflows/appengine-metadata-flexible.yaml:33]: Can you fetch `github.event.pull_request.head.sha`?

**Review by tmatsuo — CHANGES_REQUESTED** [review]

File: .github/workflows/appengine-metadata-flexible.yaml (hunk 2 above)
Comment (sofisl) [reply .github/workflows/appengine-metadata-flexible.yaml:33]: Same as other one - created an issue #2598

**Review by tmatsuo — APPROVED** [review]

**Pull request merged**